mkosi/config.py

```diff
@@ -0,0 +1,275 @@
+import argparse
+import configparser
+import dataclasses
+import enum
+import os
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Callable, Optional, Type, Union
+
+from mkosi.backend import Distribution
+from mkosi.log import die
+
+
+def parse_boolean(s: str) -> bool:
+    "Parse 1/true/yes/y/t/on as true and 0/false/no/n/f/off/None as false"
+    s_l = s.lower()
+    if s_l in {"1", "true", "yes", "y", "t", "on"}:
+        return True
+
+    if s_l in {"0", "false", "no", "n", "f", "off"}:
+        return False
+
+    raise ValueError(f"Invalid literal for bool(): {s!r}")
+
+
+def parse_source_target_paths(value: str) -> tuple[Path, Optional[Path]]:
+    src, _, target = value.partition(':')
+    if target and not Path(target).absolute():
+        die("Target path must be absolute")
+    return Path(src), Path(target) if target else None
+
+
+def config_parse_string(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    setattr(namespace, dest, value)
+
+
+def config_match_string(name: str, dest: str, value: str, namespace: argparse.Namespace) -> bool:
+    return bool(value == getattr(namespace, dest))
```

Comment:
The bool call here is a bit weird. I think this is rather an artifact of mypy not being able to look into `namespace`. Let's make this a type-check time thing instead of a runtime thing
```suggestion
    return cast(bool, value == getattr(namespace, dest))
```